### DISCUSSION TITLE
IO Broker

### DISCUSSION THREAD

**Discussion by Ivenzn** [OpenCCU/OpenCCU#2326] [🇩🇪 Fragen&Antworten] [answered]
Hallo zusammen,
ich habe folgende Konstellation. Über einen SmartPi erfasse ich den Überschuss aus der Photovoltaik. Über MQTT kommen die Daten in IO-Broker an. Dort läuft ein Blockly-Skrip das wiederum auf der Homematic einen HmIP-PCBS2 ansteuert. Damit schalte ich temperaturabhägnig (HmIP-STE2-PCB) einen Heizstab im Warmwasserbehälter. Nun habe ich ständig einen Duty-Cycle der immer wieder in den roten Bereich geht. Ich habe daher mal die Zentrale (Raspberry4) versetzt so dass diese nun neben den Steuergeräten installiert ist. Nützt leider auch nichts und ich bekomme ständig die Warnmeldung. Die Anlage läuft problemlos seit vielen Jahren und ich weis nun auch nicht mehr weiter. Alle Versuche die Daten über das MQTT-Protokoll zu erfassen und dann damit die Geräte anzusteuern laufen bei mir ins Leere da ich nur sehr einfache Programmierkenntnisse habe.

**Comment by jens-maus** [comment]
Du bringst wohl DutyCycle mit Carrier Sense durcheinander? Was genau meinst du konkret? Der DutyCycle ist eine rechtliche Begrenzung des maximal erlauben Funkverkehrs pro Zeiteinheit und der Carrier Sense ein Maß für die Störung auf dem verwendeten Funkband. Wenn der DutyCycle (wie du hier sagst) in den roten Bereich geht, so sorgt deine Automation wohl dafür das du zu viel, zu oft und zu schnell hintereinander irgendetwas schaltest bzw. Funktraffic erzeugst und damit eben in die gesetzliche Limitierung rennst (deshalb rot!). Da bringt es dann auch nix das Gerät näher heranzubringen oder an eine andere Stelle zu stellen. Das sind Dinge die man ggf. bei einem zu hohen CarrierSense tut, aber nicht bei einem zu hohen DutyCycle. Dort hilft nur das verhindern unnötiger Funksignale damit man nicht in den gesetzlich limitierten Wert rennt weil die 868MHz Frequenz nunmal eine gesharte Frequenz ist die der Gesetzgeber in der EU beschränkt hat damit dein Nachbar diese freie Funkfrequenz auch nutzen kann.

**Comment by MichaelN0815** [comment]
Dann zeig mal das Programm, mit dem Du das steuerst.

**Reply by Baxxy13** [reply]
Machst du jetzt auch Blockly-Scripte Micha? 😉
Meine Glaskugel sagt aber auch das hier die MQTT Daten eventuell im Sekundentakt reinkommen und der Aktor ohne Prüfung seines Zustandes einfach immer wieder angesprochen wird.

**Comment by Ivenzn** [comment]
Ich stell die Antwort auch hier rein damit auch Andere davon was haben.
Hallo Jens, hallo Michael,

vielen Dank für die rasche Antwort.

zunächst mal die Hardcopy von RaspberryMatic die hier mal gerade bei 4% ist. Das der momentan so niedrig ist liegt eventuell daran, dass gerade alle Bedingungen erfüllt sind und der Heizstab eingeschalten ist. Sonst liegt der im gelben oder sogar im roten Bereich. Dazu kommt jetzt neuerdings auch noch die Meldung "kein Internet" was ausgeschlossen werden kann da ich über einen direkten Draht zur FritzBox verfüge. Das nächste Bild stammt aus dem IO-Broker wo mit Blockly die Bedingungen zum Ein- bzw. Ausschalten definiert sind.

Das dritte Bild ist dann auch aus IO-Broker der die Daten anzeigt welche der MQTT-Broker im Sekundentakt liefert. Relevant ist hier nur "Ptot".
#

PS

Ich würde mir ja gern den Umweg über IO-Broker sparen und direkt den MQTT-Server in Homematic verwenden. Aber hier stoße ich eben an meine einfachen Programmierkenntnisse und habe dazu auch nichts in den entsprechenden Foren gefunden. Meiner Ansicht nach müsste es doch einen Weg geben über ein virtuelles Gerät die Daten zu empfangen und auszuwerten.

![grafik](https://github.com/jens-maus/RaspberryMatic/assets/70340943/45e27966-dad5-4a95-ba7c-38e9b2f63048)
![grafik](https://github.com/jens-maus/RaspberryMatic/assets/70340943/63ff2bf5-dea7-4153-b595-5f90b3f934ee)

![grafik](https://github.com/jens-maus/RaspberryMatic/assets/70340943/002d4515-75f8-48ca-9013-9fce77367f96)



Gruß

Norbert

**Reply by MichaelN0815** [reply, marked as answer]
Du solltest den Zustand des Heizstabs mit abfragen. Wenn Du laufend Schaltbefehle sendest, obwohl der Zustand längst erreicht ist, treibt das den DC eben nach oben.

**Comment by Ivenzn** [comment]
Verstehe ich jetzt nicht ganz. Wenn ich den Zustand auf true oder false abfrage, dann läuft ja auch ständig eine Anfrage auf.
Oder wäre das der Haken bei "löschen wenn läuft" ??

**Reply by jens-maus** [reply]
Nein, @MichaelN0815 meint du solltest VOR dem setzen des STATE des Heizstabes auf true/false nich eine zusätzliche Prüfung einbauen ob der STATE nicht bereits diesen Wert hat, weil dann kannst du dir das setzen des State ja sparen und sparst damit dann auch Funkverkehr, ergo DutyCycle… Und nein, beim Prüfen des State wird nicht jedesmal des Aktor per funkverkehr abgefragt…

**Reply by Ivenzn** [reply]
Ich habe es nochmal angepasst und nun ist es im grünen Bereich.
Vielen Dank für die Unterstützung.
